File: 2487-optimal-partition-of-string/2487-optimal-partition-of-string.java
class Solution {
    public int partitionString(String s) {
        List<String> arr=new ArrayList<>();
        String str="";
        for(int i=0;i<s.length();i++){
            
          System.out.println(str.contains(s.substring(i,i+1)));
            if(str.contains(s.substring(i,i+1))){
                arr.add(str);
                str=s.substring(i,i+1);
            }
            else{
                str+=s.substring(i,i+1);
            }
            
        }
        return arr.size()+1;
    }
}

File: 2487-optimal-partition-of-string/2487-optimal-partition-of-string.py
class Solution:
    def partitionString(self, s: str) -> int:
        listy=[]
        res=0
        for i in range(len(s)):
            if s[i] in listy:
                res+=1
                listy=[]
            listy.append(s[i])
        return res+1